ui/src/views/Roles/ViewRoles/index.jsx

```diff
@@ -15,21 +16,30 @@ import HelpView from '../../../components/HelpView';
   },
 }))
 export default class ViewRoles extends PureComponent {
-  state = {
-    roleSearch: '',
-  };
-
   handleRoleSearchSubmit = roleSearch => {
-    this.setState({ roleSearch });
+    const { location, history } = this.props;
+    const query = parse(location.search.slice(1));
+
+    if (query.roleSearch !== roleSearch) {
+      const newQuery = {
+        ...query,
+        roleSearch,
+      };
+
+      history.push({
+        search: stringify(newQuery, { addQueryPrefix: true }),
+      });
+    }
   };
 
   handleCreate = () => {
     this.props.history.push('/auth/roles/create');
   };
 
   render() {
-    const { classes, description } = this.props;
-    const { roleSearch } = this.state;
+    const { classes, description, location } = this.props;
+    const query = parse(location.search.slice(1));
+    const { roleSearch = '' } = query;
```

Comment:
Rename `roleSearch` to `search` for the query name (similar to the diff posted earlier).